frontend/src/common/UserDisplay.tsx

```diff
@@ -0,0 +1,93 @@
+import EmailIcon from '@mui/icons-material/Email'
+import UserIcon from '@mui/icons-material/Person'
+import { Box, Divider, Popover, Stack, Typography } from '@mui/material'
+import { useListUsers } from 'actions/user'
+import { MouseEvent, useEffect, useState } from 'react'
+import Loading from 'src/common/Loading'
+import { EntityObject } from 'types/v2/types'
+
+type ConditionalUserDisplayProps =
+  | {
+      entityId?: string
+      entity?: never
+    }
+  | {
+      entityId?: never
+      entity?: EntityObject
+    }
+
+type CommonUserDisplayProps = {
+  hidePopover?: boolean
+}
+
+type UserDisplayProps = CommonUserDisplayProps & ConditionalUserDisplayProps
+
+export default function UserDisplay({ entity, entityId, hidePopover = false }: UserDisplayProps) {
+  const [user, setUser] = useState<EntityObject | undefined>(entity || undefined)
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
+  const open = Boolean(anchorEl)
```

Comment:
This looks like a MUI example way of doing popovers. I generally prefer to do something like `const open = useMemo(() => !!anchorEl, [anchorEl])`